packages/hardhat/contracts/strategy/CrabOTC.sol

```diff
@@ -0,0 +1,212 @@
+        bytes32 s;
+    }
+
+    event DepositOTC(
+        address indexed depositor,
+        uint256 crabAmount,
+        uint256 wSqueethAmount,
+        uint256 depositedAmount,
+        uint256 executedPrice,
+        address trader
+    );
+
+    event WithdrawOTC(
+        address indexed depositor,
+        uint256 crabAmount,
+        uint256 ethSent,
+        uint256 wSqueethAmount,
+        uint256 executedPrice,
+        address trader
+    );
+
+    bytes32 private constant _CRAB_BALANCE_TYPEHASH =
+        keccak256(
+            "Order(address initiator,address trader,uint256 quantity,uint256 price,bool isBuying,uint256 expiry,uint256 nonce)"
+        );
+
+    constructor(address _crab) EIP712("CrabOTC", "2") {
+        require(_crab != address(0), "Invalid crab address");
+
+        crab = _crab;
+        weth = ICrabStrategyV2(_crab).weth();
+        controller = ICrabStrategyV2(_crab).powerTokenController();
+        wPowerPerp = ICrabStrategyV2(_crab).wPowerPerp();
+        IERC20(ICrabStrategyV2(_crab).wPowerPerp()).approve(_crab, type(uint256).max);
+    }
+
+    /**
+     * @notice set nonce to true
+     * @param _nonce the number to be set true
+     */
+    function setNonceTrue(uint256 _nonce) external {
+        nonces[msg.sender][_nonce] = true;
+    }
+
+    /**
+     * @dev Deposit into strategy by selling the minted oSqth
+     * @param _totalEth Total amount of ETH to deposit value + eth from selling minted oSqth
+     * @param _order A signed order to swap the tokens
+     */
+    function deposit(uint256 _totalEth, Order memory _order) external payable {
+        _verifyOrder(_order);
+        require(_order.isBuying, "Should be a buy order");
+
+        uint256 depositedEth = msg.value;
+
+        uint256 wSqueethQuantity = _getWSqueethToMint(_totalEth);
+        require(_order.quantity >= wSqueethQuantity, "Order quantity is less than needed");
+
+        uint256 wethAmount = wSqueethQuantity.wmul(_order.price);
+
+        IWETH9(weth).transferFrom(_order.trader, address(this), wethAmount);
+        IWETH9(weth).withdraw(wethAmount);
+
+        ICrabStrategyV2(crab).deposit{value: _totalEth}();
+        uint256 crabAmount = IERC20(crab).balanceOf(address(this));
+        IERC20(crab).transfer(_order.initiator, crabAmount);
+        IERC20(wPowerPerp).transfer(_order.trader, wSqueethQuantity);
+
+        uint256 excessEth = address(this).balance;
+
+        if (excessEth > uint256(0)) {
+            depositedEth = depositedEth.sub(excessEth);
+            payable(_order.initiator).sendValue(excessEth);
+        }
+
+        emit DepositOTC(_order.initiator, crabAmount, wSqueethQuantity, depositedEth, _order.price, _order.trader);
+    }
+
+    /**
+     * @dev Withdraw from strategy by buying osqth
+     * @param _crabAmount Amount of crab to withdraw
+     * @param _order A signed order to swap the tokens
+     */
+    function withdraw(
+        uint256 _crabAmount,
+        uint256 _maxPrice,
```

Comment:
Seems unnecessary to have a maxPrice since initiator is coming in with the order which they presumably find acceptable?